Lib/traceback.py

```diff
@@ -1081,50 +1081,33 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
 
         self._is_syntax_error = False
         self._have_exc_type = exc_type is not None
-        if exc_type is not None:
+
+        if self._have_exc_type:
             self.exc_type_qualname = exc_type.__qualname__
             self.exc_type_module = exc_type.__module__
+            if issubclass(exc_type, SyntaxError):
+                # Handle SyntaxError's specially
```

Comment:
Bad grammar (apostrophe). Does it make sense to fold this into the new function? I'd remove the compute prefix though, style in Python is to avoid things like get_, set_, etc.